strawberry/schema/execute.py

```diff
@@ -256,3 +259,135 @@ def execute_sync(
         errors=execution_context.result.errors,
         extensions=extensions_runner.get_extensions_results_sync(),
     )
+
+
+async def subscribe(
+    schema: GraphQLSchema,
+    *,
+    extensions: Sequence[Union[Type[SchemaExtension], SchemaExtension]],
+    execution_context: ExecutionContext,
+    process_errors: Callable[[List[GraphQLError], Optional[ExecutionContext]], None],
+) -> AsyncGenerator[Tuple[bool, ExecutionResult], None]:
+    """
+    The graphql-core subscribe function returns either an ExecutionResult or an
+    AsyncGenerator[ExecutionResult, None].  The former is returned in case of an error
+    during parsing or validation.
+    Because we need to maintain execution context, we cannot return an
+    async generator, we must _be_ an async generator.  So we yield a
+    (bool, ExecutionResult) tuple, where the bool indicates whether the result is an
```

Comment:
This tuple hack is kinda hard to understand... You might wanna check what I did at https://github.com/strawberry-graphql/strawberry/pull/2810/files#diff-88aa6fd17e4c6feac6e7152ebd3f2b8f972544c444a071b550e4d23061b97a3fR215 where if there is an error I return `ExecutionResultError`

Reply:
That is a good idea.  Yes, tuples are problematic and thought this might be a sticking point.  I'll create a special exception class instead, much nicer.

Reply:
Well, either an exception, or a special result class..  I think the exception might be cleaner, since one _expects_ a subscription and the failure to get one is an exception of sorts.  I'll see which one is nicer.